uniffi_bindgen/src/bindings/swift/gen_swift.rs

```diff
@@ -121,6 +121,52 @@ mod filters {
         })
     }
 
+    pub fn literal_swift(literal: &Literal) -> Result<String, askama::Error> {
+        fn typed_number(type_: &Type, num_str: String) -> Result<String, askama::Error> {
+            Ok(match type_ {
+                // special case Int32.
+                Type::Int32 => num_str,
+                // otherwise use constructor e.g. UInt8(x)
+                Type::Int8
+                | Type::UInt8
+                | Type::Int16
+                | Type::UInt16
+                | Type::UInt32
+                | Type::Int64
+                | Type::UInt64
+                | Type::Float32
+                | Type::Float64 => format!("{}({})", type_swift(type_)?, num_str),
+                _ => panic!("Unexpected literal: {} is not a number", num_str),
+            })
+        }
+
+        Ok(match literal {
+            Literal::Boolean(v) => format!("{}", v),
+            Literal::String(s) => format!("\"{}\"", s),
+            Literal::Null => "nil".into(),
+            Literal::EmptySequence => "[]".into(),
+            Literal::EmptyMap => "[:]".into(),
+            Literal::Enum(v, _) => format!(".{}", enum_variant_swift(v)?),
+            Literal::Int(i, radix, type_) => typed_number(
+                type_,
+                match radix {
+                    Radix::Octal => format!("0o{:o}", i),
+                    Radix::Decimal => format!("{}", i),
+                    Radix::Hexadecimal => format!("{:#x}", i),
+                },
+            )?,
+            Literal::UInt(i, radix, type_) => typed_number(
+                type_,
+                match radix {
+                    Radix::Octal => format!("0o{:o}", i),
+                    Radix::Decimal => format!("{}", i),
+                    Radix::Hexadecimal => format!("{:#x}", i),
+                },
```

Comment:
This level of duplication felt acceptable. 

Using the 8 byte `u64`, `i64` and `f64` as internal representations for number literals was the compromise between lots of duplication, and flexibility of representation.